runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -30,44 +30,198 @@
 #include "chpltypes.h"
 #include "error.h"
 
-void chpl_mem_layerInit(void) {
-  void* start;
-  size_t size;
 
-  chpl_comm_desired_shared_heap(&start, &size);
+//TODO move the chunk replacement code into a separate file
 
+// TODO put these into a struct?
+void*  heap_base = NULL;
```

Comment:
These seem like they ought to be file static.  If they can't be, they need a chpl_mem_ prefix.